craft_application/commands/lifecycle.py

```diff
@@ -364,6 +366,9 @@ def _run(
             raise RuntimeError(f"Step name {step_name} passed to pack command.")
         super()._run(parsed_args, step_name="prime")
 
+        self._services.package.update_project()
```

Comment:
so all told this is the third place we have this logic right? PrimeCommand, here, and in snapcraft's PackCommand? Do we have a plan to refactor this in the (near) future?

Reply:
How about this?  
I de-duped the code here (see the recent commit) and I will be able to drop it from snapcraft in https://github.com/canonical/snapcraft/pull/4797